lib/hanami/cli/registry.rb

```diff
@@ -0,0 +1,46 @@
+require "hanami/cli/command_registry"
+
+module Hanami
+  class Cli
+    module Registry
+      def self.extended(base)
+        base.class_eval do
+          @commands = CommandRegistry.new
+        end
+      end
+
+      def register(name, command = nil, aliases: [])
+        if block_given?
+          yield Prefix.new(@commands, name, aliases)
+        else
+          command.command_name = name if Cli.command?(command)
```

Comment:
@timriley I'm concerned about this problem too.

We have two ways:

  1. Pass the name to the initializer, but that means we loose the feature to have any callable object as command. Because the following syntax will blow up: `register "foo", ->(*) { ... }`.
  1. Mutate the class ivar only if the command is a `Hanami::Cli::Command`.

---

In my understanding you need for `dry` the ability to register any callable object as a command.

While this is nice-to-have feature, I've got the feeling that people interested into `hanami-cli` will "buy" the entire experience: both the registry **and** the commands.

So if you need it, we'll keep it, otherwise we can assume that all the registered commands are `Hanami::Cli::Command` instances, so then we can pass the name to the initializer.

Please let us to know. 😄